integration-cli/docker_api_containers_test.go

```diff
@@ -1636,3 +1636,26 @@ func (s *DockerSuite) TestPostContainerStop(c *check.C) {
 		c.Fatal(err)
 	}
 }
+
+// #14170
+func (s *DockerSuite) TestPostContainersCreateWithStringOrSliceEntrypoint(c *check.C) {
+	config := struct {
+		Image      string
+		Entrypoint string
+		Cmd        []string
+	}{"busybox", "echo", []string{"hello", "world"}}
+	_, _, err := sockRequest("POST", "/containers/create?name=echotest", config)
+	c.Assert(err, check.IsNil)
+	out, _ := dockerCmd(c, "start", "-a", "echotest")
+	c.Assert(strings.TrimSpace(out), check.Equals, "hello world")
+
+	config2 := struct {
+		Image      string
+		Entrypoint []string
+		Cmd        []string
```

Comment:
I'd intentionally left off the cmd integration test since the semantics for that are different... However I have this now and will force push when I have a chance.
Unit test for cmd is already in place.